debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/macho/MachoDumpReader.java

```diff
@@ -357,18 +361,15 @@ public long getProcessId()
 
 	public List<? extends IOSThread> getThreads()
 	{
-		List<ThreadState> x86ThreadStates = new ArrayList<>(dumpFile.threads.states.size());
-		for(ThreadState state : dumpFile.threads.states) {
-			if (state.flavor == ThreadCommand.x86_THREAD_STATE) {
-				x86ThreadStates.add(state);
-			} else if (_signalNumber == -1 && state.flavor == ThreadCommand.x86_EXCEPTION_STATE) {
-				_signalNumber = state.registers.get("err").intValue();
+		List<IOSThread> threads = new ArrayList<>(dumpFile.threads.size());
+		for (ThreadCommand command : dumpFile.threads) {
+				OSXThread nativeThread = new OSXThread(0, command);
+				threads.add(nativeThread);
+				if (_signalNumber == -1) {
+					// will use the first thread's exception values, which should be the primary thread
+					_signalNumber = nativeThread.registers.get("err").intValue();
```

Comment:
What if `nativeThread.registers` has no `err` key?

Reply:
I believe every Mac OSX machine has a `x86_EXCEPTION_STATE` so that key will always exist. It should produce an exception otherwise.

Reply:
We should do better than `NullPointerException`, though.